kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -434,7 +435,11 @@ private String getCommitFilename(String encodedPartition) {
   }
 
   private String fileKey(String topicsPrefix, String keyPrefix, String name) {
-    return topicsPrefix + dirDelim + keyPrefix + dirDelim + name;
+    String base = "";
+    if (StringUtils.isNotBlank(topicsPrefix) && !topicsPrefix.equals("/")) {
+      base = topicsPrefix + dirDelim;
+    }
+    return base + keyPrefix + dirDelim + name;
```

Comment:
Also, I apologise for being overly specific here, but this might be one of the rare cases where a sightly lengthy ternary operator might be more readable than a block of 3-4 lines. 

How about?

```java
private String fileKey(String topicsPrefix, String keyPrefix, String name) {
    String suffix = keyPrefix + dirDelim + name;
    return StringUtils.isNotBlank(topicsPrefix)
           ? topicsPrefix + dirDelim + suffix
           : suffix;
}
```